index.js

```diff
@@ -79,9 +85,16 @@ async function assumeRole(params) {
   }
 
   let assumeFunction = sts.assumeRole.bind(sts);
+  
+  // These are customizations needed for the GH OIDC Provider
+  if(isDefined(webIdentityToken)) {
+    delete assumeRoleRequest.Tags;
 
-  if(isDefined(webIdentityTokenFile)) {
-    core.debug("webIdentityTokenFile provided. Will call sts:AssumeRoleWithWebIdentity and take session tags from token contents.")
+    assumeRoleRequest.WebIdentityToken = webIdentityToken;
+    assumeRoleRequest.DurationSeconds = 3600;
```

Comment:
We would still want the ability to set `role-duration-seconds` using the input. There are a few roles for us that I specifically would want to intentionally shorten this value. I think a default of 3600 for OIDC to match the role default in AWS makes a lot of sense, so the normal path is to not have to worry about setting anything. I think this should probably just be toggling the default value based on `useGitHubOIDCProvider()` type logic near the definition. Conditional defaults may be a little odd, but I think you're right about wanting to maintain backwards compatibility, and I think this lines up with the new documentation.

Maybe something like `const roleDurationSeconds = core.getInput('role-duration-seconds', {required: false}) || getDefaultRoleDurationSeconds(???);`?

Reply:
Again, another good catch. I shouldn't have tried to write this code so early on a Monday.

Reply:
What are your thoughts on
```javascript
    if(useGitHubOIDCProvider()) {
      webIdentityToken = await getWebIdentityToken(maskAccountId, region);
      roleDurationSeconds = core.getInput('role-duration-seconds', {required: false}) || DEFAULT_ROLE_DURATION_FOR_OIDC_ROLES;
```

Then I removed the manual change I just made. My only challenge with the code you provided is that I'd have to pass a bunch of env variables into the getDefaultRoleSDurationSeconds() which may not have been set yet so the behavior of the program might change based on where the call was made from, which seemed a bit hard to maintain.

Reply:
That looks great to me.